lib/klaviyo/oauth2.ex

```diff
@@ -0,0 +1,45 @@
+defmodule Klaviyo.OAuth2 do
+  alias Klaviyo.RequestOperation
+
+  @spec authorize_url(String.t(), Enum.t()) :: String.t()
+  def authorize_url(url \\ "https://www.klaviyo.com/oauth/authorize", params) do
+    params =
+      params
+      |> put_in([:response_type], "code")
+      |> put_in([:code_challenge_method], "S256")
+
+    url
+    |> URI.parse()
+    |> Map.put(:query, URI.encode_query(params))
+    |> URI.to_string()
+  end
+
+  @spec code_challenge(String.t()) :: String.t()
+  def code_challenge(code_verifier) do
+    digest = :crypto.hash(:sha256, code_verifier)
+
+    Base.encode64(digest, padding: false)
+  end
+
+  @spec code_verifier(pos_integer) :: String.t()
+  def code_verifier(length \\ 128) do
```

Comment:
minor: This should never be less than 48 characters. Would it make sense to add a guard and throw an error if that was supplied? maybe an overkill so feel free to ignore.